src/napari/layers/shapes/shapes.py

```diff
@@ -2747,43 +2747,52 @@ def _update_thumbnail(self, event=None):
 
             self.thumbnail = colormapped
 
-    def remove_selected(self) -> None:
-        """Remove any selected shapes."""
-        index = list(self.selected_data)
-        to_remove = sorted(index, reverse=True)
+    def remove(self, indices: list[int]) -> None:
+        """Remove any shapes at the given indices.
+
+        Parameters
+        ----------
+        indices : List[int]
+            List of indices of shapes to remove from the layer.
+        """
+        to_remove = sorted(indices, reverse=True)
 
-        if len(index) > 0:
+        if len(indices) > 0:
             self.events.data(
                 value=self.data,
                 action=ActionType.REMOVING,
                 data_indices=tuple(
-                    index,
+                    indices,
                 ),
                 vertex_indices=((),),
             )
             for ind in to_remove:
                 self._data_view.remove(ind)
 
-            self._feature_table.remove(index)
-            self.text.remove(index)
+            self._feature_table.remove(indices)
```

Comment:
```suggestion
            self.selected_data = self.selected_data - set(indices)
            self._feature_table.remove(indices)
```
Ensure we deselect the Shapes, so there is no issue with highlights/outlines.

I'm not sure this is the best place for this line, but I think it makes sense? And it fixes the issue I was having